server/src/services/Analysis.ts

```diff
@@ -1,29 +1,45 @@
+import { type } from 'os';
+
 class Analysis {
   average: number;
   median: number;
   standardDeviation: number;
 
-  constructor(array: number[]) {
-    this.average = array.reduce((a, b) => a + b, 0) / array.length;
-    this.median = this.countMedian(array);
-    this.standardDeviation = this.countStandardDeviation(array);
+  constructor(array: string[]) {
+    const fitered = this.checkArray(array);
+
+    this.average = this.countAverage(fitered);
+    this.median = this.countMedian(fitered);
+    this.standardDeviation = this.countStandardDeviation(fitered);
   }
   countMedian(array: number[]) {
     const mid = Math.floor(array.length / 2),
       nums = [...array].sort((a, b) => a - b);
     return array.length % 2 !== 0 ? nums[mid] : (nums[mid - 1] + nums[mid]) / 2;
   }
   countStandardDeviation(array: number[]) {
+    const avg = this.countAverage(array);
```

Comment:
Dlaczego liczysz średnią drugi raz? Masz własciwoś average na obiekcie już, szanujmy moce obliczeniowe :D

Reply:
Bo sypie się wtedy error w chuj średnia jest undefined, czy to może być przez hoisting?